CHANGELOG.md

```diff
@@ -6,11 +6,159 @@ document.
+  the `len` method as well as the type definition.
+  [#6853](https://github.com/rust-lang/rust-clippy/pull/6853)
+* [`let_underscore_drop`]: Only lints on types which implement `Drop`
+  [#6682](https://github.com/rust-lang/rust-clippy/pull/6682)
+* [`unit_arg`]: No longer lints on unit arguments when they come from a path expression.
+  [#6601](https://github.com/rust-lang/rust-clippy/pull/6601)
+* [`cargo_common_metadata`]: No longer lints if 
+  [`publish = false`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-publish-field)
+  is defined in the manifest
+  [#6650](https://github.com/rust-lang/rust-clippy/pull/6650)
+
+### Suggestion Fixes/Improvements
+
+* [`collapsible_match`]: Fixed lint message capitalization
+  [#6766](https://github.com/rust-lang/rust-clippy/pull/6766)
+* [`or_fun_call`]: Improved suggestions for `or_insert(vec![])`
+  [#6790](https://github.com/rust-lang/rust-clippy/pull/6790)
+* [`manual_map`]: No longer expands macros in the suggestions
+  [#6801](https://github.com/rust-lang/rust-clippy/pull/6801)
+* Aligned Clippy's lint messages with the rustc dev guide
+  [#6787](https://github.com/rust-lang/rust-clippy/pull/6787)
+
+### ICE Fixes
+
+* [`zero_sized_map_values`]
+  [#6866](https://github.com/rust-lang/rust-clippy/pull/6866)
+
+### Documentation Improvements
+
+* [`useless_format`]: Improved the documentation example
+  [#6854](https://github.com/rust-lang/rust-clippy/pull/6854)
+* Clippy's [`README.md`]: Includes a new subsection on running Clippy as a rustc wrapper 
+  [#6782](https://github.com/rust-lang/rust-clippy/pull/6782)
+
+### Others
+
+* Clippy's lintcheck tool:
+  * The log now includes stats about changed lint counts
+    [#6800](https://github.com/rust-lang/rust-clippy/pull/6800)
+  * A custom list of crates can be defined using the `LINTCHECK_TOML` environment value
+    [#6735](https://github.com/rust-lang/rust-clippy/pull/6735)
+  * Support parallel crate checking using rayon
+    [#6764](https://github.com/rust-lang/rust-clippy/pull/6764)
+  * Collect ICEs, prevent a log override for custom crate lists and support local sources
+    [#6708](https://github.com/rust-lang/rust-clippy/pull/6708)
+  * Fully rerun lintcheck if Clippy was rebuilt since last log update
+    [#6780](https://github.com/rust-lang/rust-clippy/pull/6780)
+  * Support git sources
+    [#6686](https://github.com/rust-lang/rust-clippy/pull/6686)
+* Internal refactorings:
+  * Extracted Clippy's `utils` module into the new `clippy_utils` crate
+    [#6756](https://github.com/rust-lang/rust-clippy/pull/6756)
+  * Refactor `loops.rs` file into `loops` module
+    [#6824](https://github.com/rust-lang/rust-clippy/pull/6824)
+  * Moved lints in methods module into their own modules
+    [#6826](https://github.com/rust-lang/rust-clippy/pull/6826)
+  * Refactored the `transmute.rs` file into a new `transmute` module
+    [#6716](https://github.com/rust-lang/rust-clippy/pull/6716)
```

Comment:
I wouldn't put those here. The changelog is only for user facing stuff. The point about extracting `utils` is fine, because people might want to use the clippy utils.

Reply:
Notes for next time :upside_down_face: